rtengine/simpleprocess.cc

```diff
@@ -1361,58 +1367,47 @@ class ImageProcessor {
                 }
             }
 
-            delete tmplab;
-            tmplab = cropped;
-            cx = 0;
-            cy = 0;
-
+            tmplab.swap(cropped);
             params.crop.enabled = false;
         }
 
         assert(params.resize.enabled);
 
         // resize image
-        LabImage *resized = new LabImage(imw, imh);
-        ipf.Lanczos(tmplab, resized, tmpScale);
-        delete tmplab;
-        tmplab = resized;
+        {
+            std::unique_ptr<LabImage> resized(new LabImage(imw, imh));
+            ipf.Lanczos(tmplab.get(), resized.get(), scale_factor);
+            tmplab.swap(resized);
```

Comment:
Sure. :smile: 

`swap()` is meant to exchange two things. For `std::unique_ptr<>` this means, both of them live on. You let the old instance die afterwards, I know, but that's not how it was implemented before:

```c++
delete tmplab;
tmplab = resized;
```

The equivalent with `std::unique_ptr<>` is:

```c++
tmplab = std::move(resize);
```

I know, this may seem like nitpicking to you, but consider that a destructor might have side-effects, or that the memory is released earlier. Both `swap` and `move` say what they do, and whenever I read `swap` I'll ask myself, why the writer preserved the other instance and what's next with it.

Best,
Flössie